synapse/config/server.py

```diff
@@ -666,6 +750,48 @@ def default_config(self, server_name, data_dir_path, **kwargs):
         # 'true'.
         #
         #show_users_in_user_directory: false
+
+        # Message retention policy at the server level.
+        #
+        # Room admins and mods can define a retention period for their rooms using the
+        # 'im.vector.room.retention' state event, and server admins can cap this period
+        # by setting the 'min_lifetime' and 'max_lifetime' config options. This feature is
+        # disabled by default.
+        #
+        # If this feature is enabled, Synapse will regularly look for and purge events
+        # which are older than the room's maximum retention period. Synapse will also
+        # filter events received over federation so that events that should have been
+        # purged are ignored and not stored again.
+        #
+        # Rooms that lack the 'im.vector.room.retention' state event will be considered as
+        # if their maximum retention period is equal to the 'max_lifetime' parameter from
+        # the server's configuration.
+        #
+        # Server admins can also define the settings of the background jobs purging the
+        # events which lifetime has expired under the 'purge_jobs' section.
+        # If no configuration is provided, a single job will be setup to delete expired
+        # events in every room daily.
+        # Each job's configuration defines which range of message lifetime the job takes
+        # care of. For example, if 'min_lifetime' is '2d' and 'max_lifetime' is '3d', the
+        # job will handle purging expired events in rooms which state defines a
+        # 'max_lifetime' that's between 2 and 3 days.
+        #
+        # The rationale for this per-job configuration is that server admins might want to
```

Comment:
having `min_lifetime` and `max_lifetime` at both the top level and per-job feels confusing. Probably they just need renaming to better reflect their purposes.